File: lexer-java/src/main/java/basicplus/frontend/LexerError.java
// ============================================================
// LexerError.java
// Diagnóstico (error o aviso) producido durante la tokenización.
// ============================================================
package basicplus.frontend;

public final class LexerError {
    public final String message;
    public final int line;
    public final int column;

    public LexerError(String message, int line, int column) {
        this.message = message;
        this.line = line;
        this.column = column;
    }

    @Override
    public String toString() {
        return "[" + line + ":" + column + "] error léxico: " + message;
    }
}

File: lexer/LexerError.cs
// ============================================================
// LexerError.cs
// Diagnóstico (error o aviso) producido durante la tokenización.
// ============================================================

namespace BasicPlus.Lexer;

public sealed class LexerError
{
    public string Message { get; }
    public int Line { get; }
    public int Column { get; }

    public LexerError(string message, int line, int column)
    {
        Message = message;
        Line = line;
        Column = column;
    }

    public override string ToString() => $"[{Line}:{Column}] error léxico: {Message}";
}
